pyspark_dubber/sql/functions/array.py

```diff
@@ -0,0 +1,31 @@
+import ibis
+
+from pyspark_dubber.sql import DataFrame
+from pyspark_dubber.sql.expr import Expr, LiteralValue
+from pyspark_dubber.sql.functions.normal import ColumnOrName
+
+def array_append(self,
+    col: ColumnOrName,
+    value: LiteralValue | Expr
+) -> "DataFrame":
+  if isinstance(col, str):
+    col_expr = self[col]
+  else:
+    col_name = col.get_name()
+    col_expr = col
+  if isinstance(value, LiteralValue):
+    value = ibis.literal(value)
+  new_col = col_expr.concat(ibis.array([value]))
+  return self.mutate(**{col_name: new_col})
```

Comment:
Should return an expressions using these I guess:

- https://ibis-project.org/reference/expression-collections#ibis.expr.types.arrays.ArrayValue.concat
- https://ibis-project.org/reference/expression-collections#ibis.array